### PR TITLE
fix: add null check for select options in table widget

### LINKED ISSUE #24857: Editing select column type while adding a new row breaks table widget TypeError: Cannot read properties of null (reading 'label')
Sentry Issue: [APPSMITH-67Y](https://appsmith.sentry.io/issues/4107010204/?referrer=github_integration)

```
TypeError: Cannot read properties of null (reading 'label')
  at _this.renderSingleSelectItem (widgets/SelectWidget/component/index.tsx:157:51)
  at <object>.children (widgets/SelectWidget/component/index.tsx:239:10)
...
(11 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by keyurparalkar**
## Description
This PR passes a `[]` to the select component when some values in the select options that are passed to the select cell component are `null`.

This fixes issue where the table widget breaks when select options for the select column type contains `{{null}}` or `{{[null, null]}}`

#### PR fixes following issue(s)
Fixes #24857

#### Type of change

- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?

- [x] Manual
    - To test that table widget does not break when `{{[null]}}`, `{{[null, null]}}`, `{{null}}` is passed to select options or new row select options.
    - To test that table widget does not break when select options or new row select options contains API binding that does not have any data 
- [ ] Jest
- [x] Cypress
    - should test that select column dropdown has no results found string when select option is {{null}}
    - should test that select column dropdown has no results found string when select option is {{[null, null]}}  

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/widgets/TableWidgetV2/component/cellComponents/SelectCell.tsx [hunk 1]
```diff
@@ -189,6 +190,8 @@ export const SelectCell = (props: SelectProps) => {
     .map((d: DropdownOption) => d.value)
     .indexOf(value);
 
+  const isOptionsNull = props.options && props.options.some((d) => isNil(d));
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidgetV2/component/cellComponents/SelectCell.tsx:193]: we need to do this in the validation itself.

File: app/client/src/widgets/TableWidgetV2/widget/propertyUtils.ts [hunk 2]
```diff
@@ -899,6 +899,15 @@ export function selectColumnOptionsValidation(
               uniqueValues = new Set();
 
               for (let j = 0; j < value[i].length; j++) {
+                if (_.isNil(value[i][j])) {
+                  _isValid = false;
+                  _message = `${generateErrorMessagePrefix(
+                    i,
+                    j,
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidgetV2/widget/propertyUtils.ts:906]: Just move this error message to a constant inside the function and use it in both places.

**Review by sbalaji1192 — COMMENTED** [review]
Please go ahead with QA. will give my approval before merging.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/select_spec_2.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  entityExplorer,
+  propPane,
+  agHelper,
+  draggableWidgets,
+  deployMode,
+  table,
+  locators,
+} from "../../../../../../support/Objects/ObjectsCore";
+
+const TABLE_DATA_1 = `[
```
Comment (sbalaji1192) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/select_spec_2.ts:11]: we have a function called AddSampleData to add this to the table in table helper

**Comment by chandannkumar** [comment]
@keyurparalkar Can you create a DP for this PR

**Comment by chandannkumar** [comment]
Tested this PR and LGTM

**Review by sbalaji1192 — APPROVED** [review]

**Pull request merged**